pkg/apis/org/v1/che_types.go

```diff
@@ -457,6 +457,21 @@ type CheClusterSpecAuth struct {
 	// Identity provider container custom settings.
 	// +optional
 	IdentityProviderContainerResources ResourcesCustomSettings `json:"identityProviderContainerResources,omitempty"`
+	// Enables native user mode. Currently works only on OpenShift and DevWorkspace engine.
+	// Native User mode uses OpenShift OAuth directly as identity provider, without Keycloak.
+	// +optional
+	NativeUserMode *bool `json:"nativeUserMode,omitempty"`
+	// Gateway sidecar responsible for authentication when NativeUserMode is enabled.
+	// See link:https://github.com/oauth2-proxy/oauth2-proxy[oauth2-proxy] or link:https://github.com/openshift/oauth-proxy[openshift/oauth-proxy].
+	// +optional
+	GatewayAuthenticationSidecarImage string `json:"gatewayAuthenticationSidecarImage,omitempty"`
```

Comment:
Would it make sense to add the `*Image` config options to checluster v2 and modify the v1<->v2 conversion, too?

Reply:
I don't know how to do that. Is v2 in che-operator already?